vsock_proxy/src/starter.rs

```diff
@@ -45,17 +45,35 @@ pub fn check_allowlist(
             .as_vec()
             .ok_or("No allowlist field")?;
 
+        // Obtain the remote server's IP address.
+        let mut addrs = Proxy::parse_addr(&remote_host, only_4, only_6)
+            .map_err(|err| format!("Could not parse remote address: {}", err))?;
+        let remote_addr = *addrs.get(0).ok_or("No IP address found")?;
+
         for raw_service in services {
+            let addr = raw_service["address"].as_str().ok_or("No address field")?;
             let port = raw_service["port"]
                 .as_i64()
                 .ok_or("No port field or invalid type")?;
             let port = port as u16;
 
-            let addr = raw_service["address"].as_str().ok_or("No address field")?;
-            let addrs = Proxy::parse_addr(addr, only_4, only_6)?;
+            // Start by matching against ports.
+            if port != remote_port {
+                continue;
+            }
+
+            // Attempt to match directly against the allowlisted hostname first.
+            if addr == remote_host {
+                info!("Matched with host name \"{}\" and port \"{}\"", addr, port);
+                return Ok(remote_addr);
+            }
+
+            // If hostname matching failed, attempt to match against IPs.
+            addrs = Proxy::parse_addr(addr, only_4, only_6)?;
             for addr in addrs.into_iter() {
-                if addr == remote_addr && port == remote_port {
-                    return Ok(());
+                if addr == remote_addr {
+                    info!("Matched with host IP \"{}\" and port \"{}\"", addr, port);
```

Comment:
wouldn't this be too verbose ? doesn't it get printed for every packet ?

Reply:
That only happens when the vsock-proxy is launched, and if RUST_LOG=info is set. So it shouldn't get printed for every packet.